packages/xchain-ethereum/src/client.ts

```diff
@@ -339,41 +271,46 @@ export default class Client implements XChainClient, EthereumClient {
    * Get the ETH balance of a given address.
    *
    * @param {Address} address By default, it will return the balance of the current wallet. (optional)
-   * @returns {Array<ETHBalance>} The all balance of the address.
+   * @returns {Array<Balances>} The all balance of the address.
+   *
+   * @throws {"Invalid asset"} throws when the give asset is an invalid one
    */
-  getBalance = async (address?: Address): Promise<Balances> => {
+  getBalance = async (address?: Address, asset?: Asset): Promise<Balances> => {
     try {
-      if (!this.ethplorerApiKey) {
-        return Promise.reject(new Error('Missing API Key for Ethplorer'))
-      }
       address = address || this.getAddress()
-      const account = await ethplorerAPI.getAddress(this.getEthplorerUrl(), address, this.ethplorerApiKey)
-      const balances: Balances = [
-        {
-          asset: AssetETH,
-          amount: assetToBase(assetAmount(account.ETH.balance, ETH_DECIMAL)),
-        },
-      ]
-
-      if (account.tokens) {
-        account.tokens.forEach((token) => {
-          const decimals = parseInt(token.tokenInfo.decimals)
-          const { symbol, address: tokenAddress } = token.tokenInfo
-          if (validateSymbol(symbol) && this.validateAddress(tokenAddress)) {
-            const tokenAsset = assetFromString(`${ETHChain}.${symbol}-${tokenAddress}`)
-            if (tokenAsset) {
-              balances.push({
-                asset: tokenAsset,
-                amount: baseAmount(token.balance, decimals),
-              })
-            }
-          }
-        })
-      }
 
-      return balances
+      if (asset && assetToString(asset) !== assetToString(AssetETH)) {
+        // Handle token balances
+        const assetAddress = getTokenAddress(asset)
+        if (!assetAddress) {
+          throw new Error('Invalid asset')
+        }
+
+        const balance = await etherscanAPI.getTokenBalance(
+          this.etherscan.baseUrl,
+          address,
+          assetAddress,
+          this.etherscan.apiKey,
+        )
+        const decimal = await this.call<BigNumberish>(assetAddress, erc20ABI, 'decimals', [])
```

Comment:
Since [`decimals` method is optional](https://eips.ethereum.org/EIPS/eip-20#methods), use its returns value or `18` (ETH_DECIMAL)